In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt


Read Dataset

In [4]:
df=pd.read_csv("/content/drive/MyDrive/winequality_red (1).csv")
df.head()

fixed acidity  volatile acidity  citric acid  ...  sulphates  alcohol  quality
0            7.4              0.70         0.00  ...       0.56      9.4        5
1            7.8              0.88         0.00  ...       0.68      9.8        5
2            7.8              0.76         0.04  ...       0.65      9.8        5
3           11.2              0.28         0.56  ...       0.58      9.8        6
4            7.4              0.70         0.00  ...       0.56      9.4        5

[5 rows x 12 columns]

In [7]:
df.tail()

fixed acidity  volatile acidity  citric acid  ...  sulphates  alcohol  quality
1594            6.2             0.600         0.08  ...       0.58     10.5        5
1595            5.9             0.550         0.10  ...       0.76     11.2        6
1596            6.3             0.510         0.13  ...       0.75     11.0        6
1597            5.9             0.645         0.12  ...       0.71     10.2        5
1598            6.0             0.310         0.47  ...       0.66     11.0        6

[5 rows x 12 columns]

In [5]:
df.describe()

fixed acidity  volatile acidity  ...      alcohol      quality
count    1599.000000       1599.000000  ...  1599.000000  1599.000000
mean        8.319637          0.527821  ...    10.422983     5.636023
std         1.741096          0.179060  ...     1.065668     0.807569
min         4.600000          0.120000  ...     8.400000     3.000000
25%         7.100000          0.390000  ...     9.500000     5.000000
50%         7.900000          0.520000  ...    10.200000     6.000000
75%         9.200000          0.640000  ...    11.100000     6.000000
max        15.900000          1.580000  ...    14.900000     8.000000

[8 rows x 12 columns]

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1599 entries, 0 to 1598
Data columns (total 12 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   fixed acidity         1599 non-null   float64
 1   volatile acidity      1599 non-null   float64
 2   citric acid           1599 non-null   float64
 3   residual sugar        1599 non-null   float64
 4   chlorides             1599 non-null   float64
 5   free sulfur dioxide   1599 non-null   float64
 6   total sulfur dioxide  1599 non-null   float64
 7   density               1599 non-null   float64
 8   pH                    1599 non-null   float64
 9   sulphates             1599 non-null   float64
 10  alcohol               1599 non-null   float64
 11  quality               1599 non-null   int64  
dtypes: float64(11), int64(1)
memory usage: 150.0 KB


In [9]:
from sklearn.model_selection import train_test_split
X=df.loc[:,df.columns!='quality']
Y=df['quality']
xtrain,xtest,ytrain ,ytest  = train_test_split(X,Y,test_size=0.30,random_state=42)

In [12]:
xtest.head()

fixed acidity  volatile acidity  citric acid  ...    pH  sulphates  alcohol
803             7.7              0.56         0.08  ...  3.24       0.66      9.6
124             7.8              0.50         0.17  ...  3.39       0.48      9.5
350            10.7              0.67         0.22  ...  3.28       0.98      9.9
682             8.5              0.46         0.31  ...  3.33       0.54      9.8
1326            6.7              0.46         0.24  ...  3.39       0.60     10.6

[5 rows x 11 columns]

In [13]:
print(f'xtrain shape {xtrain.shape} ')
print(f'ytrain shape {ytrain.shape} ')
print(f'xtest shape {xtest.shape} ')
print(f'ytest shape {ytest.shape}')

xtrain shape (1119, 11) 
ytrain shape (1119,) 
xtest shape (480, 11) 
ytest shape (480,)


In [15]:
from sklearn.neural_network import MLPClassifier
model=MLPClassifier(hidden_layer_sizes=(150,100,45),activation='relu',max_iter=300,solver='adam')
model.fit(xtrain,ytrain)

MLPClassifier(hidden_layer_sizes=(150, 100, 45), max_iter=300)

In [16]:
ypred=model.predict(xtest)


In [18]:
from sklearn.metrics import accuracy_score
accuracy =accuracy_score(ytest,ypred)
print('accuracy of model using relu is :',accuracy)

accuracy of model using relu is : 0.5875


Accuracy of model using relu and adam is 0.5875



In [33]:
from sklearn.neural_network import MLPClassifier
model1=MLPClassifier(hidden_layer_sizes=(150,100,45),activation='tanh',max_iter=300,solver='adam')
model1.fit(xtrain,ytrain)
ypred=model1.predict(xtest)

/usr/local/lib/python3.7/dist-packages/sklearn/neural_network/_multilayer_perceptron.py:696: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (300) reached and the optimization hasn't converged yet.
  ConvergenceWarning,


In [34]:
from sklearn.metrics import accuracy_score
accuracy =accuracy_score(ytest,ypred)
print('accuracy of model1 using tanh is :',accuracy)

accuracy of model1 using tanh is : 0.5895833333333333


Accuracy of model using tanh and adam is 0.5895833333333333

In [35]:
from sklearn.neural_network import MLPClassifier
model2=MLPClassifier(hidden_layer_sizes=(150,100,45),activation='logistic',max_iter=300,solver='adam')
model2.fit(xtrain,ytrain)
ypred=model2.predict(xtest)

/usr/local/lib/python3.7/dist-packages/sklearn/neural_network/_multilayer_perceptron.py:696: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (300) reached and the optimization hasn't converged yet.
  ConvergenceWarning,


In [36]:
from sklearn.metrics import accuracy_score
accuracy =accuracy_score(ytest,ypred)
print('accuracy of model2 using logistics is :',accuracy)

accuracy of model2 using logistics is : 0.5666666666666667


Accuracy of model using logistic and adam is 0.5666666666666667

In [37]:
from sklearn.neural_network import MLPClassifier
model3=MLPClassifier(hidden_layer_sizes=(150,100,45),activation='relu',max_iter=300,solver='sgd')
model3.fit(xtrain,ytrain)
ypred=model3.predict(xtest)

In [38]:
from sklearn.metrics import accuracy_score
accuracy =accuracy_score(ytest,ypred)
print('accuracy of model3 using relu and sgd is :',accuracy)

accuracy of model3 using relu and sgd is : 0.49583333333333335


Accuracy of model using relu and sgd is 0.49583333333333335

In [39]:
from sklearn.neural_network import MLPClassifier
model4=MLPClassifier(hidden_layer_sizes=(150,100,45),activation='tanh',max_iter=300,solver='sgd')
model4.fit(xtrain,ytrain)
ypred=model4.predict(xtest)

In [43]:
from sklearn.metrics import accuracy_score
accuracy =accuracy_score(ytest,ypred)
print('accuracy of model4 using tanh and sgd is :',accuracy)

accuracy of model4 using tanh and sgd is : 0.40625


Accuracy of model using tanh and sgd is  0.40625

In [41]:
from sklearn.neural_network import MLPClassifier
model5=MLPClassifier(hidden_layer_sizes=(150,100,45),activation='logistic',max_iter=300,solver='sgd')
model5.fit(xtrain,ytrain)
ypred=model5.predict(xtest)

In [42]:
from sklearn.metrics import accuracy_score
accuracy =accuracy_score(ytest,ypred)
print('accuracy of model5 using logistic and sgd is :',accuracy)

accuracy of model5 using logistic and sgd is : 0.40625


accuracy of model5 using logistic and sgd is : 0.40625